### PR TITLE
BUG: Fix regression when using Series with arrow string array

### LINKED ISSUE #51844: BUG: Passing pyarrow string array + dtype to `pd.Series` throws ArrowInvalidError on 2.0rc
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd, pyarrow as pa

pd.Series(pa.array("the quick brown fox".split()), dtype="string")

# ArrowInvalid: Needed to copy 1 chunks with 0 nulls, but zero_copy_only was True
```

### Issue Description

The example above errors when it probably shouldn't. Here's the example + traceback:

```python
import pandas as pd, pyarrow as pa

pd.Series(pa.array("the quick brown fox".split()), dtype="string")
# ArrowInvalid: Needed to copy 1 chunks with 0 nulls, but zero_copy_only was True
```

<details>
<summary> traceback </summary>

```pytb
---------------------------------------------------------------------------
ArrowInvalid                              Traceback (most recent call last)
Cell In[1], line 3
      1 import pandas as pd, pyarrow as pa
----> 3 pd.Series(pa.array("the quick brown fox".split()), dtype="string")

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/core/series.py:490, in Series.__init__(self, data, index, dtype, name, copy, fastpath)
    488         data = data.copy()
    489 else:
--> 490     data = sanitize_array(data, index, dtype, copy)
    492     manager = get_option("mode.data_manager")
    493     if manager == "block":

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/core/construction.py:565, in sanitize_array(data, index, dtype, copy, allow_2d)
    563     _sanitize_non_ordered(data)
    564     cls = dtype.construct_array_type()
--> 565     subarr = cls._from_sequence(data, dtype=dtype, copy=copy)
    567 # GH#846
    568 elif isinstance(data, np.ndarray):

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/core/arrays/string_.py:359, in StringArray._from_sequence(cls, scalars, dtype, copy)
    355     result[na_values] = libmissing.NA
    357 else:
    358     # convert non-na-likes to str, and nan-likes to StringDtype().na_value
--> 359     result = lib.ensure_string_array(scalars, na_value=libmissing.NA, copy=copy)
    361 # Manually creating new array avoids the validation step in the __init__, so is
    362 # faster. Refactor need for validation?
    363 new_string_array = cls.__new__(cls)

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/_libs/lib.pyx:712, in pandas._libs.lib.ensure_string_array()

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/_libs/lib.pyx:754, in pandas._libs.lib.ensure_string_array()

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pyarrow/array.pxi:1475, in pyarrow.lib.Array.to_numpy()

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pyarrow/error.pxi:100, in pyarrow.lib.check_status()

ArrowInvalid: Needed to copy 1 chunks with 0 nulls, but zero_copy_only was True
```

</details>

This also occurs if `dtype="string[pyarrow]"`:

```python
pd.Series(pa.array("the quick brown fox".split()), dtype="string[pyarrow]")
# ArrowInvalid: Needed to copy 1 chunks with 0 nulls, but zero_copy_only was True
```

<details>
<summary> traceback </summary>

```pytb
---------------------------------------------------------------------------
ArrowInvalid                              Traceback (most recent call last)
Cell In[2], line 3
      1 import pandas as pd, pyarrow as pa
----> 3 pd.Series(pa.array("the quick brown fox".split()), dtype="string[pyarrow]")

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/core/series.py:490, in Series.__init__(self, data, index, dtype, name, copy, fastpath)
    488         data = data.copy()
    489 else:
--> 490     data = sanitize_array(data, index, dtype, copy)
    492     manager = get_option("mode.data_manager")
    493     if manager == "block":

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/core/construction.py:565, in sanitize_array(data, index, dtype, copy, allow_2d)
    563     _sanitize_non_ordered(data)
    564     cls = dtype.construct_array_type()
--> 565     subarr = cls._from_sequence(data, dtype=dtype, copy=copy)
    567 # GH#846
    568 elif isinstance(data, np.ndarray):

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/core/arrays/string_arrow.py:149, in ArrowStringArray._from_sequence(cls, scalars, dtype, copy)
    146     return cls(pa.array(result, mask=na_values, type=pa.string()))
    148 # convert non-na-likes to str
--> 149 result = lib.ensure_string_array(scalars, copy=copy)
    150 return cls(pa.array(result, type=pa.string(), from_pandas=True))

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/_libs/lib.pyx:712, in pandas._libs.lib.ensure_string_array()

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pandas/_libs/lib.pyx:754, in pandas._libs.lib.ensure_string_array()

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pyarrow/array.pxi:1475, in pyarrow.lib.Array.to_numpy()

File ~/miniconda3/envs/pandas-2.0/lib/python3.10/site-packages/pyarrow/error.pxi:100, in pyarrow.lib.check_status()

ArrowInvalid: Needed to copy 1 chunks with 0 nulls, but zero_copy_only was True

```

</details>

### Expected Behavior

This probably shouldn't error, and instead result in a series of the appropriate dtype.

I would note that this seems to work on 1.5.3

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 1a2e300170efc08cb509a0b4ff6248f8d55ae777
python           : 3.10.9.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Tue Jun 21 20:50:28 PDT 2022; root:xnu-7195.141.32~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 2.0.0rc0
numpy            : 1.23.5
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 67.4.0
pip              : 23.0.1
Cython           : None
pytest           : 7.2.1
hypothesis       : None
sphinx           : 6.1.3
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.11.0
pandas_datareader: None
bs4              : 4.11.2
bottleneck       : 1.3.7rc1
brotli           : None
fastparquet      : None
fsspec           : 2023.1.0
gcsfs            : None
matplotlib       : 3.7.0
numba            : 0.56.4
numexpr          : 2.8.4
odfpy            : None
openpyxl         : 3.1.1
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2023.1.0
scipy            : 1.10.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : None
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #51844 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -752,8 +752,11 @@ cpdef ndarray[object] ensure_string_array(
             out = arr.astype(str).astype(object)
             out[arr.isna()] = na_value
             return out
-
-        arr = arr.to_numpy()
+        if hasattr(arr, "type"):
+            # pyarrow array
+            arr = np.array(arr)
```
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:757]: 1) can we handle this at a higher level?  2) why doesnt the pyarrow object's to_numpy work here?

Comment (phofl) [reply pandas/_libs/lib.pyx:757]: It sets zero_copy_only to True which causes raising here.

we have to do a check like this somewhere as long as arrow is not a hard dependency, can do it at a higher level as well

Comment (phofl) [reply pandas/_libs/lib.pyx]: moved it to a higher level, not sure if it is better though

**Review by mroeschke — APPROVED** [review]

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Comment by jbrockmendel** [comment]
thanks @phofl

File: pandas/core/arrays/string_.py [hunk 2]
```diff
@@ -352,6 +352,9 @@ def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy: bool = Fal
             result[na_values] = libmissing.NA
 
         else:
+            if hasattr(scalars, "type"):
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/string_.py:355]: Why can't we do an actual isinstance check?

If we have some helper method `is_pyarrow_array` like the following, this should be quite easy? 

```
def is_pyarrow_array(obj):
    if _pyarrow_installed:
        return isinstance(obj, pa.Array)
    return False
```

with variables `_pyarrow_installed` and `pa` filled in once on first import.

Comment (jorisvandenbossche) [reply pandas/core/arrays/string_.py:355]: PR that does this -> https://github.com/pandas-dev/pandas/pull/52830

**Pull request merged**